tests/phpunit/tests/media.php

```diff
@@ -2293,6 +2293,20 @@ public function test_wp_filter_content_tags_srcset_sizes_with_preexisting_srcset
 		$this->assertSame( $img, wp_filter_content_tags( $img ) );
 	}
 
+	/**
+	 * @ticket 55347
+	 *
+	 * @covers ::wp_filter_content_tags
+	 */
+	public function test_wp_filter_content_tags_has_filter() {
+		$img = get_image_tag( self::$large_id, '', '', '', 'medium' );
+
+		add_filter( 'wp_img_tag_add_adjust', static function( $content ) {
+			return $content . 'data-test="filter_active"';
+		} );
+
+		$this->assertStringContainsString( 'data-test="filter_active"', wp_filter_content_tags( $img ), 'The content filter should have been applied.' );
```

Comment:
There is a helper for ensuring a particular action runs.

```suggestion
		$filter = new MockAction();
		add_filter( 'wp_img_tag_add_adjust', array( &$filter, 'filter' ) );
		$img_tag_1 = get_image_tag( self::$large_id, '', '', '', 'medium' );

		wp_filter_content_tags( $img_tag_1 );
		$this->assertSame( 1, $filter->get_call_count() );
```